eclair-core/src/main/scala/fr/acinq/eclair/channel/Helpers.scala

```diff
@@ -775,12 +789,12 @@ object Helpers {
           withTxGenerationLog("claim-htlc-timeout", logSuccess = false) {
             Transactions.makeClaimHtlcTimeoutTx(remoteCommitTx.tx, outputs, commitments.localParams.dustLimit, localHtlcPubkey, remoteHtlcPubkey, remoteRevocationPubkey, commitments.localParams.defaultFinalScriptPubKey, add, feeratePerKwHtlc, commitments.commitmentFormat)
           }.map(claimHtlcTx => {
-            claimHtlcTx.input.outPoint -> withTxGenerationLog("claim-htlc-timeout") {
+            Some(claimHtlcTx.input.outPoint -> withTxGenerationLog("claim-htlc-timeout") {
               val sig = keyManager.sign(claimHtlcTx, keyManager.htlcPoint(channelKeyPath), remoteCommit.remotePerCommitmentPoint, TxOwner.Local, commitments.commitmentFormat)
               Right(Transactions.addSigs(claimHtlcTx, sig))
-            }
+            })
           })
-      }.toSeq.flatten.toMap
+      }.toSeq.flatten.flatten.toMap
```

Comment:
Especially here, the `.flatten.flatten` is a bit weird, leveraging `collect` and pattern matching may be more clear?